Makefile

```diff
@@ -9,7 +9,7 @@ VERSION=$(shell cat ./version)
 # Change DUMMY_PROVIDER below to reflect the name of the service under development.  The
 # value of this variable is used in LOCAL_LOCATION, and is also used in the
 DUMMY_PROVIDER=metal
-LOCAL_LOCATION=~/.local/share/terraform/plugins/terraform.example.com/$(DUMMY_PROVIDER)/hpegl/$(VERSION)/linux_amd64/
+LOCAL_LOCATION=~/.local/share/terraform/plugins/terraform.example.com/$(DUMMY_PROVIDER)/hpegl/$(VERSION)/linux_$(GOARCH)
```

Comment:
Where is GOARCH being set from? This package can be built from other sources so you can't assume that environment variables will already be set.

Reply:
1. GOARCH is set to "PF_ARM" or "PF_AMD" in the script "./script/ci-build".
2. These variables "PF_ARM" and "PF_AMD" is defined in ./script/make.sh" which is getting called on every fresh build "make prod-portal" 

Code Snippet for reference
===================================================
File: $QUAKE/script/ci-build

```
# Project qcli build and copy the artifact.
# Linux
echo "LINUX-ONLY: building ${BIN_NAME} at location ${dir}."
(cd ${dir}; GOOS=$PF_OS GOARCH=$PF_AMD make build)
...
# MAC
echo "MAC-ONLY: building ${BIN_NAME} at location ${dir}."
(cd ${dir}; GOOS=$PF_OS GOARCH=$PF_ARM make build)

```
File: $QUAKE/script/make.sh

```
PLATFORM=`uname -m`

export PF_OS='linux'
export PF_ARM='arm64'
export PF_AMD='amd64'

if [ "$PLATFORM" == "aarch64" ]; then
    export SIM="true"
    export PF_ARCH="arm64"
elif [ "$PLATFORM" == "x86_64" ]; then
    export SIM=""
    export PF_ARCH="amd64"
else
    echo "OS/Platform not supported"
fi
```

Reply:
Like the other PR. People that are building the terraform provider might not be building from inside quake. This repo needs to be able to be built by itself.

Reply:
I agree with Neil, this is an "indipendent repo". Can we get the `GOARCH` value form the result of go env?

Reply:
@neilgierman  , @dbozzato81 
1. By default this Makefile only builds and generates Linux/amd64 based terraform provider binary.
2. To avoid this limitation (and make it generic):
2A. From inside QUAKE, I'm calling the make build as below:
- For Linux:
`(cd ${dir}; CIRCLE_TAG=${repo_desired_tag} **GOOS=$PF_OS** **GOARCH=$PF_AMD** make build)`
- For Mac:
`(cd ${dir}; CIRCLE_TAG=${repo_desired_tag} **GOOS=$PF_OS** **GOARCH=$PF_ARM** make build)`
2B. From outside QUAKE, we can call the make build as below:
(cd ${dir}; CIRCLE_TAG=${repo_desired_tag} **GOOS=`go env GOOS`** **GOARCH=`go env GOARCH** make build)

Let me further look into it and revert.

Reply:
Sharing further findings:
1. Inside Quake, as per default design, we do "make build" TERRAFORM PROVIDER BINARY for LINUX as well as MAC
- For Linux, it generates a binary (linux/amd64)
- For Mac, it generates the same binary (linux/amd64) - failed on Mac
so, we need to remove/modify hardcoded "amd64" in the repo (./hpegl-metal-terraform-resources/Makefile)

For this, I've tried to make it generic using this PR.
Please suggest.

Reply:
This does not appear to be a backwards compatible change. This is forcing everyone who builds from outside of quake environment to pass in a new var GOARCH. Need to define GOARCH with default value matching existing default (amd64), then override if user provides different value.

Reply:
Also, as others have pointed out, this is a public repo that anyone can clone & build. There must not be any references to INTERNAL PROJECT NAMEs. I'm going to delete some of the comments. Please be mindful of this.